File: Strings/pagina05/2336/2336.php
<?php 

$mod = 1000000007;

while($abc = readline()){

    $resultado = 0;
    for($i = 0; $i < strlen($abc); $i++){
        $code = ord($abc[$i]) - ord("A");
        $resultado = ($resultado * 26 + $code) % $mod;
    }

    echo "$resultado\n";
}

File: Strings/__2000__/2336/2336.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        long MOD = 1000000007;

        while (input.hasNextLine()) {
            String abc = input.nextLine().trim();
            long resultado = 0;

            for (char chr : abc.toCharArray()) {
                int code = chr - 'A';
                resultado = (resultado * 26 + code) % MOD;
            }

            System.out.println(resultado);
        }

        input.close();
    }
}
